### PR TITLE
Added `NID_pkcs9_contentType` and `ub_` to compatibility layer

### PR DESCRIPTION
**Pull request by dgarske**
ZD 11742

### CODE REVIEW

File: wolfssl/openssl/asn1.h [hunk 1]
```diff
@@ -92,6 +92,17 @@
 #define ASN1_STRING_FLAG_MSTRING         0x040
 #define ASN1_STRING_FLAG_EMBED           0x080
 
+/* X.509 PKI size limits from RFC2459 (appendix A) */
+/* internally our limit is CTC_NAME_SIZE (64) - overriden with WC_CTC_NAME_SIZE */
+#define ub_name                    CTC_NAME_SIZE /* 32768 */
```
Comment (SparkiDev) [inline wolfssl/openssl/asn1.h:97]: What are the numbers in the comments at the end of the lines?

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: wolfssl/openssl/asn1.h (hunk 1 above)
Comment (dgarske) [reply wolfssl/openssl/asn1.h:97]: Those are the RFC defined max limits (commented for reference), but each field is limited by CTC_NAME_SIZE, which can be build-time overridden.

**Comment by dgarske** [comment]
Awaiting customer feedback on these, not ready for merge.

**Comment by dgarske** [comment]
> Awaiting customer feedback on these, not ready for merge.

Customer is happy with change. Rebased to resolve merge conflicts. Ready for review.

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**